bgpd/bgp_zebra.c

```diff
@@ -394,12 +396,23 @@ static int bgp_interface_address_delete(int command, struct zclient *zclient,
 			   ifc->ifp->name, buf);
 	}
 
+	/* Update the router id if the interface address being deleted is
+	 * same as the bgp router id
+	 */
+	p = ifc->address;
+	if ((p->family == AF_INET) &&
+		IPV4_ADDR_SAME(&bgp->router_id, &p->u.prefix4))
```

Comment:
This would need to be `bgp->router_id_zebra` and not `bgp->router_id`. But please see my other comment above, I think all changes done in this file can be dropped.